modules/apps/portal-template/portal-template-react-renderer-impl/src/main/java/com/liferay/portal/template/react/renderer/internal/ReactRendererUtil.java

```diff
@@ -104,41 +104,36 @@ public static void renderReact(
 	}
 
 	private static void _renderJavaScript(
-			ComponentDescriptor componentDescriptor, Map<String, Object> props,
-			HttpServletRequest httpServletRequest, String placeholderId,
-			Portal portal, Writer writer)
-		throws IOException {
+		ComponentDescriptor componentDescriptor, Map<String, Object> props,
+		HttpServletRequest httpServletRequest, String placeholderId,
+		JSModuleLauncher jsModuleLauncher, Portal portal, Writer writer) {
 
-		StringBundler dependenciesSB = new StringBundler(7);
+		StringBundler javascriptSB = new StringBundler(15);
 
-		dependenciesSB.append(componentDescriptor.getModule());
-		dependenciesSB.append(" as renderFunction");
-		dependenciesSB.append(placeholderId);
+		javascriptSB.append("Liferay.Loader.require(['");
```

Comment:
All these changes look super super scary 😂

Reply:
Can you walk me through this change? How come this won't affect existing react components not-yet-migrated?

Reply:
That's part of what `ScriptData` was doing before, but it seems I've badly broken it (as seen in my local portal :-( ). So I need to check if it is because of this or the latest changes to the API...

The good thing about that change is that the code is so sensitive that if it works it must be correct :joy:

Reply:
Turned out it was the database, that needed to be recreated to avoid a NPE that was causing a bad rendering. 

Now, to the change thing: this code is the same `ScriptData` used to exectute to load AMD loader modules. It's not very complex and given that it will disappear when we remove AMD loader, I don't think it's a problem to have it duplicated.

Also, `Liferay.Loader.require` is almost an API, so we could in theory write things like this in more than one place of the portal (no need to make sure that everything passes through `ScriptData`)